docs/3_troubleshooting.md

```diff
@@ -5,13 +5,17 @@ In case you encounter any issues while using ReVanced Manager, please refer to t
 - 💉 Patching fails with an error
 
   Make sure ReVanced Manager is up to date by following [🔄 Updating ReVanced Manager](2_3_updating.md) and select the **Default** button when choosing patches.
-  
+
 - 🚫 App not installed as package conflicts with an existing package
 
   An existing installation of the app you're trying to patch is conflicting with the patched app. Uninstall the existing app before installing the patched app.
 
+  This may also occur when trying to install the previous version of an app. Android sometimes blocks downgrading apps. However, if you uninstall the existing app, and try to install the downgrade again, it usually works.
+
 - ❗️ Error code `135`, `139` or `1` when patching the app
 
+  You may be trying to patch a split APK. This is common with the YouTube app. You can visit APKMirror to download an alternate version of the app that is not split (the `bundle` label means it is split) and install it.
```

Comment:
How can the user tell whether the APK they have is a split APK? If we don't explain how to make this determination via APKMirror, then we should provide some other explanation.

> It is the user's responsibility to source APK files, a precondition also stated in the documentation of ReVanced Manager.

Sorry, I looked for the place in the documentation where the requirement to use non-split APKs was mentioned, but I didn't manage to find it. Would you mind pointing me in the right direction?

Reply:
> How can the user tell whether the APK they have is a split APK?

You can add and reference a footnote to the term "split" to an external source in a similar fashion to how Wikipedia does. The determination should not be made using APKMirror. It may be a popular source of APK binaries but it is not connected with ReVanced whatsoever, whereas the official explanation of what a split APK is, is on the Android documentation website.

> where the requirement to use non-split APKs was mentioned

You may have misunderstood what I said with:

> It is the user's responsibility to source APK files, a precondition also stated in the documentation of ReVanced Manager.

Source: https://github.com/ReVanced/revanced-manager/blob/main/docs/2_1_patching.md#user-content-fn-1-0509e8ec511256471426023611fdf9d4 (first footnote)

This is not about split APK files. Split APK files can be patched in certain circumstances.

Reply:
> We should not mention any specific apps because ReVanced is agnostic, so the documentation must also apply to any app.

Okay, I've updated the documentation to not mention YouTube, although I do feel that it would be helpful for ReVanced to document the requirements for what versions of apps it is able to patch (e.g., only the non-split version of YouTube can be patched, at least according to [this comment](https://old.reddit.com/r/revancedapp/comments/xna2gn/youtube_unsupported_version_and_split_apk/ipsg1b4/)).

> You can add and reference a footnote to the term "split" to an external source in a similar fashion to how Wikipedia does

Okay, I've updated the documentation to use a footnote. Currently, it links to a Reddit post. I am happy to update this to point at some official Android documentation about how to tell whether an APK file is split or not, if any such documentation exists. I wasn't able to find any when I searched.

> The determination should not be made using APKMirror

I'd love to know how the determination should be made, if not that way. This would surely be helpful information to other users, as well.

> Split APK files can be patched in certain circumstances.

It would be very helpful to know what circumstances these are.

Reply:
> It would be very helpful to know what circumstances these are.

Primarily, patching resources limit us from properly supporting split APK file patching. Regardless, if you mount a resource-patched APK file, it may still work. In some specific known cases, either patching fails or the app crashes regardless of mounting.

Reply:
> I'd love to know how the determination should be made, if not that way. This would surely be helpful information to other users, as well.

The split APK architecture is simple. It works similarly to the dex architecture, where Android iterates over the entries in the zip file to search for a class that is yet to be loaded in a classloader. Android queries Similarily, the internals of an APK in multiple APK files. This means a split architecture is not identifiable in itself as they are regular APK files. What gives it away is, if the APK files miss required files such as dex or resource files. In this case, this implies wether an APK file is a split or not. Each APK file also has to include a manifest, which is encoded but contains the information regarding this as well. Sites such as APKMirror distribute proprietary formats for splits that archive the separate APK files in a single file intended to be installed with their proprietary installers.